src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/dto/GitLabPersonalAccessTokenListResponseDTO.java

```diff
@@ -6,30 +6,10 @@
 
 /**
  * Stores necessary information for a GitLab response to list personal access tokens.
+ *
+ * @param id The id of the personal access token.
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public class GitLabPersonalAccessTokenListResponseDTO {
-
-    /**
-     * The id of the personal access token.
-     */
-    @JsonProperty
-    private Long id;
-
-    public GitLabPersonalAccessTokenListResponseDTO() {
-        // default constructor for Jackson
-    }
-
-    public GitLabPersonalAccessTokenListResponseDTO(Long id) {
-        this.id = id;
-    }
-
-    public Long getId() {
-        return id;
-    }
-
-    public void setId(Long id) {
-        this.id = id;
-    }
+public record GitLabPersonalAccessTokenListResponseDTO(@JsonProperty Long id) {
```

Comment:
Why do we need `@JsonProperty` here if we don't pass a different json-key value?

Reply:
just reused from the original class based implementation. Can probably be removed, but we would need to test the change